NSubstitute.DbConnection/NSubstitute.DbConnection.Dapper.Tests/DapperQueryTests.cs

```diff
@@ -199,6 +199,20 @@ public void DapperAlwaysPassesParametersToStoredProcedureQuery()
             result.Should().Throw<NotSupportedException>().WithMessage(noMatchingQueryErrorMessage);
         }
 
+        [Test]
+        public void ShouldUseSuppliedMatcherOverDefaultQueryMatching()
+        {
+            var mockConnection = Substitute.For<IDbConnection>().SetupCommands();
+            mockConnection.SetupQuery("select * from table where id = @id")
```

Comment:
I would have WithMatcher as an exteniosn on the connection, like SetupQuery is, because here the query you set it up with actually just gets ignored. Either an overload of SetupQuery(Func<string, bool>), or call it something different like SetupQueryMatching(Func<string, bool>)

I was also thinking an overload of SetupQuery taking a Regex would be good.

btw, predicate on the command is a nice idea but the command object is itself an incomplete mock so allowing people to create an arbitrary predicate on the command is likely to lead to confusion, as well as tying the implementation down to support the existing behaivour, which is why I prefer a predicate on just the query text